### ISSUE TITLE
SearchValues<T> in .NET 8

### ISSUE THREAD

**Issue by znakeeye** [benjamin-hodgson/Pidgin#156]
Thanks for your awesome library!

Not sure if the [SearchValues&lt;T&gt;](https://learn.microsoft.com/en-us/dotnet/api/system.buffers.searchvalues-1?view=net-8.0) is applicable for `Pidgin`, but it certainly sounds like a good fit.

Also, see this [this](https://endjin.com/blog/2024/01/dotnet-8-searchvalues-string-search-performance-boost) great blog post where the author tries out this new feature.

**Comment by benjamin-hodgson** [comment]
I'll need to think about this. It seems `SearchValues` is designed with string- or array-searching in mind. In general we don't have the full text available when parsing (it's streamed) so it may not be a good fit. Besides, recursive descent parsing is generally about _matching the current token_, not _searching for a specific substring_.

However I can envision a use case like "search forward in the stream using `SearchValues` and begin parsing from there". At present you need to wind forward token by token to find a match so perhaps there's hay to be made by vectorising that search. The partial data issue is a problem there though (when doing substring matching and not single-token matching).

**Comment by znakeeye** [comment]
Makes sense. I mostly thought of `AnyCharExcept()` when I stumbled upon that `SearchValues` feature.
